constraint/pkg/client/drivers/local/driver.go

```diff
@@ -0,0 +1,435 @@
+
+	return res, t, err
+}
+
+func (d *Driver) Query(ctx context.Context, target string, constraints []*unstructured.Unstructured, review interface{}, opts ...drivers.QueryOpt) ([]*types.Result, *string, error) {
+	if len(constraints) == 0 {
+		return nil, nil, nil
+	}
+
+	constraintsByKind := toConstraintsByKind(constraints)
+
+	traceBuilder := strings.Builder{}
+	constraintsMap := drivers.KeyMap(constraints)
+	path := []string{"hooks", "violation[result]"}
+
+	var results []*types.Result
+
+	// Round-trip review through JSON so that the review object is round-tripped
+	// once per call to Query instead of once per compiler.
+	reviewMap, err := toInterfaceMap(review)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for kind, kindConstraints := range constraintsByKind {
+		compiler := d.compilers.getCompiler(target, kind)
+		if compiler == nil {
+			// The Template was just removed, so ignore these Constraints.
+			continue
+		}
+
+		// Parse input into an ast.Value to avoid round-tripping through JSON when
+		// possible.
+		parsedInput, err := toParsedInput(kindConstraints, reviewMap)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		resultSet, trace, err := d.eval(ctx, compiler, path, parsedInput, opts...)
+		if err != nil {
+			resultSet = make(rego.ResultSet, 0, len(kindConstraints))
+			for _, constraint := range kindConstraints {
+				resultSet = append(resultSet, rego.Result{
+					Bindings: map[string]interface{}{
+						"result": map[string]interface{}{
+							"msg": err.Error(),
+							"key": map[string]interface{}{
+								"kind": constraint.GetKind(),
+								"name": constraint.GetName(),
+							},
+						},
+					},
+				})
+			}
+		}
+		if trace != nil {
+			traceBuilder.WriteString(*trace)
+		}
+
+		kindResults, err := drivers.ToResults(constraintsMap, resultSet)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		results = append(results, kindResults...)
+	}
+
+	traceString := traceBuilder.String()
+	if len(traceString) != 0 {
+		return results, &traceString, nil
+	}
+
+	return results, nil, nil
+}
+
+func (d *Driver) Dump(ctx context.Context) (string, error) {
+	dt := make(map[string]map[string]rego.ResultSet)
+
+	compilers := d.compilers.list()
+	for targetName, targetCompilers := range compilers {
+		targetData := make(map[string]rego.ResultSet)
+
+		for kind, compiler := range targetCompilers {
+			rs, _, err := d.eval(ctx, compiler, []string{"data"}, nil)
+			if err != nil {
+				return "", err
+			}
+			targetData[kind] = rs
+		}
+
+		dt[targetName] = targetData
+	}
+
+	resp := map[string]interface{}{
+		"data": dt,
+	}
+
+	b, err := json.MarshalIndent(resp, "", "   ")
+	if err != nil {
+		return "", err
+	}
+
+	return string(b), nil
+}
+
+// templateLibPrefix returns the new lib prefix for the libs that are specified in the CT.
+func templateLibPrefix(name string) string {
+	return fmt.Sprintf("libs.%s", name)
+}
+
+// createTemplatePath returns the package path for a given template: templates.<target>.<name>.
```

Comment:
nit: comment appears to be stale (path has changed)

Reply:
Fixed.